src/hb-set.hh

```diff
@@ -266,9 +266,9 @@ struct hb_set_t
   {
     if (unlikely (hb_object_is_immutable (this)))
       return;
-    population = 0;
-    page_map.resize (0);
-    pages.resize (0);
+
+    if (resize (0))
+      population = 0;
```

Comment:
Please make clear() always succeed (even if set is in error state), like it used to.